lib/src/execution/windows/execution.cc

```diff
@@ -358,6 +358,19 @@ namespace facter { namespace execution {
         option_set<execution_options> const& options,
         uint32_t timeout)
     {
+        // Since we use a job object in the windows world, we want to
+        // be sure we're not in a job object, or at least able to
+        // break our processes out if we are in one.
+        BOOL in_job = FALSE;
+        IsProcessInJob(GetCurrentProcess(), nullptr, &in_job);
+        if (in_job) {
+            JOBOBJECT_BASIC_LIMIT_INFORMATION limits;
+            QueryInformationJobObject(nullptr, JobObjectBasicLimitInformation, &limits, sizeof(limits), nullptr);
+            if (!(limits.LimitFlags & JOB_OBJECT_LIMIT_BREAKAWAY_OK)) {
```

Comment:
Here's what I'd propose:

* Query to see if the current process is either not part of a job or part of a job and that `JOB_OBJECT_LIMIT_BREAKAWAY_OK` is set on the job.  If so, set flags to `CREATE_NO_WINDOW | CREATE_BREAKAWAY_FROM_JOB`.  Otherwise, just `CREATE_NO_WINDOW`.
* If `CREATE_BREAKAWAY_FROM_JOB` is set in the creation flags, create a new job and store in `hJob` and associate the child process with the job.  Otherwise, leave `hJob` initialized to simply `scoped_resource<HANDLE>(nullptr)`.
* The call to TerminateProcess should only occur if `hJob` is not nullptr.

Thoughts?